### PR TITLE
Enable tiered projections for GpuProjectExec

### PR DESCRIPTION
**Pull request by jbrennan333**
Signed-off-by: Jim Brennan <jimb@nvidia.com>

Change `GpuProjectExec` to use tiered projections if `spark.rapids.sql.tiered.project.enabled=true`, which is the default setting.

In testing with NDS2 power run, there was no measurable performance improvement or degradation with this change.   But it is possible that some real-world queries may benefit, and this will also help to shake out any issues with tiered projections.

During the NDS2 power run at scale 100, I observed 21874 invocations of GpuProjectExec, of which only 43 resulted in more than one tier.  A projection with one tier is equivalent to just doing a normal non-tiered projection.
I also validated that the output was correct for the power run.

### CODE REVIEW

**Comment by jbrennan333** [comment]
Closes https://github.com/NVIDIA/spark-rapids/issues/6528.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala [hunk 1]
```diff
@@ -344,6 +353,17 @@ case class GpuProjectAstExec(
     // Process tiers sequentially
     recurse(exprTiers, inputAttrTiers, batch, true)
   }
+
+  def tieredProjectAndClose(cb: ColumnarBatch, opTime: GpuMetric): ColumnarBatch = {
+    val nvtxRange = new NvtxWithMetrics("ProjectExec", NvtxColor.CYAN, opTime)
```
Comment (abellina) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala:358]: nit, we could use `withResource` here for both `nvtxRange` and `cb`

**Review by abellina — COMMENTED** [review]
LGTM, I had one nit.

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala (hunk 1 above)
Comment (jbrennan333) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala:358]: Good suggestion.

**Comment by jbrennan333** [comment]
I am not seeing a failure when I run generate_expr_test.py locally.
Will try to build again.

**Comment by jbrennan333** [comment]
build

**Comment by jbrennan333** [comment]
build

**Comment by jbrennan333** [comment]
Merged up to branch-22.12.  Will run build again.

**Comment by jbrennan333** [comment]
build

**Review by abellina — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Comment by jbrennan333** [comment]
Thanks for the reviews!

**Pull request merged**